### DISCUSSION TITLE
when validatePageNumber, pageNumber == nextPageNumber，throw Exception，then do  create newNextPage fail

### DISCUSSION THREAD

**Discussion by LuvianKindred** [jahlborn/jackcess#16] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/fce7642d-66ce-4365-adb1-c18508eed8e8)
when data do in the lastPage, need create newNextPage,but aways happen pageNumber == nextPageNumber, it's seem create newNextPage fail, now nextPageNumber is 8018, but do next try, nextPageNumber is 8019 and also create newNextPage fail

so why pageNumber have to  greater than nextPageNumber? please

**Comment by jahlborn** [comment]
i'm not sure i have enough context to answer your question.  typically, when a new page is created, the size of the channel should be extended so that this check should not fail for the recently allocated page number.  however, i can't say more than that without knowing the context of what is calling validatePageNumber().  have you encountered a bug in jackcess?

note, i prefer to keep jackcess discussions in the sourceforge project https://sourceforge.net/p/jackcess/discussion/456474/ .

**Reply by LuvianKindred** [reply]
![image](https://github.com/user-attachments/assets/3e80630f-c0dd-4806-b892-cd3b2e997d99)
![image](https://github.com/user-attachments/assets/97c03ba6-8b3f-4cc7-884e-3a00f4fd3eac)
exception stackTrace

**Reply by jahlborn** [reply]
those line numbers don't seem to align with the latest code.  what version of jackcess are you using?

**Reply by jahlborn** [reply, marked as answer]
I'm guessing that you are using an outdated version of Jackcess and that updating to the latest version would fix your issue.
